core/src/main/java/org/elasticsearch/transport/netty/MessageChannelHandler.java

```diff
@@ -226,14 +226,17 @@ protected String handleRequest(Channel channel, Marker marker, StreamInput buffe
         transportServiceAdapter.onRequestReceived(requestId, action);
         NettyTransportChannel transportChannel = null;
         try {
-            transport.inFlightRequestsBreaker().addEstimateBytesAndMaybeBreak(messageLengthBytes, "<transport_request>");
-            transportChannel = new NettyTransportChannel(transport, transportServiceAdapter, action, channel,
-                requestId, version, profileName, messageLengthBytes);
-
             final RequestHandlerRegistry reg = transportServiceAdapter.getRequestHandler(action);
             if (reg == null) {
                 throw new ActionNotFoundTransportException(action);
             }
+            long reservedBytes = 0;
+            if (reg.isCheckSizeLimit()) {
```

Comment:
Without digging into it it sounds like `addWithoutBreaking` is what we want for those requests? They still take up space so why not count them? I get that this could push us past the limit but that seems, if not a good thing, at least appropriate.